src/yxy.rs

```diff
@@ -0,0 +1,305 @@
+
+///CIE 1931 Yxy (xyY) with an alpha component. See the [`Yxya` implementation in `Alpha`](struct.Alpha.html#Yxya).
+pub type Yxya<T = f32> = Alpha<Yxy<T>, T>;
+
+///The CIE 1931 Yxy (xyY)  color space.
+///
+///Yxy is a luminance-chromaticity color space derived from the CIE XYZ
+///color space. It is widely used to define colors. The chromacity diagrams
+///for the color spaces are a plot of this color space's x and y coordiantes.
+///
+///Conversions and operations on this color space assumes the CIE Standard
+///Illuminant D65 as the white point, and the 2° standard colorimetric
+///observer.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Yxy<T: Float = f32> {
+
+    ///x chromacity co-ordinate derived from XYZ color space as X/(X+Y+Z).
+    ///Typical range is between 0 and 1
+    pub x: T,
+
+    ///y chromacity co-ordinate derived from XYZ color space as Y/(X+Y+Z).
+    ///Typical range is between 0 and 1
+    pub y: T,
+
+    ///luma (Y) was a measure of the brightness or luminance of a color.
+    ///It is the same as the Y from the XYZ color space. Its range is from
+    ///0 to 1, where 0 is black and 1 is white.
+    pub luma: T,
+}
+
+impl<T: Float> Yxy<T> {
+    ///CIE Yxy.
+    pub fn new(x: T, y: T, luma: T,) -> Yxy<T> {
+        Yxy {
+            x: x,
+            y: y,
+            luma: luma,
+        }
+    }
+}
+
+///<span id="Yxya"></span>[`Yxya`](type.Yxya.html) implementations.
+impl<T: Float> Alpha<Yxy<T>, T> {
+    ///CIE Yxy and transparency.
+    pub fn new(x: T, y: T, luma: T, alpha: T) -> Yxya<T> {
+        Alpha {
+            color: Yxy::new(x, y, luma),
+            alpha: alpha,
+        }
+    }
+}
+
+impl<T: Float> Limited for Yxy<T> {
+    fn is_valid(&self) -> bool {
+        self.x >= T::zero() && self.x <= T::one() &&
+        self.y >= T::zero() && self.y <= T::one() &&
+        self.luma >= T::zero() && self.luma <= T::one()
+    }
+
+    fn clamp(&self) -> Yxy<T> {
+        let mut c = *self;
+        c.clamp_self();
+        c
+    }
+
+    fn clamp_self(&mut self) {
+        self.x= clamp(self.x, T::zero(), T::one());
+        self.y = clamp(self.y, T::zero(), T::one());
+        self.luma = clamp(self.luma, T::zero(), T::one());
+    }
+}
+
+impl<T: Float> Mix for Yxy<T> {
+    type Scalar = T;
+
+    fn mix(&self, other: &Yxy<T>, factor: T) -> Yxy<T> {
+        let factor = clamp(factor, T::zero(), T::one());
+
+        Yxy {
+            x: self.x + factor * (other.x - self.x),
+            y: self.y + factor * (other.y - self.y),
+            luma: self.luma + factor * (other.luma - self.luma),
+        }
+    }
+}
+
+impl<T: Float> Shade for Yxy<T> {
+    type Scalar = T;
+
+    fn lighten(&self, amount: T) -> Yxy<T> {
+        Yxy {
+            x: self.x,
+            y: self.y,
+            luma: self.luma + amount,
+        }
+    }
+}
+
+impl<T: Float> Default for Yxy<T> {
+    fn default() -> Yxy<T> {
+        Yxy::new(T::zero(), T::zero(), T::zero())
```

Comment:
I was thinking about this. Rgb's are using black as the default (0,0,0) . Lab is using 0,0,0 as well which translates to rgb 0's. both yxy and xyz as 0's translate to rgb 0's. So 0's might be a better value for these